agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/BaseDecorator.java

```diff
@@ -97,20 +92,23 @@ public Span onPeerConnection(final Span span, final InetSocketAddress remoteConn
 
   public Span onPeerConnection(final Span span, final InetAddress remoteAddress) {
     assert span != null;
-    String hostName = remoteAddress.getHostName();
-    if (!hostName.equals(remoteAddress.getHostAddress())) {
-      span.setAttribute(SemanticAttributes.NET_PEER_NAME.key(), remoteAddress.getHostName());
-    }
-    span.setAttribute(SemanticAttributes.NET_PEER_IP.key(), remoteAddress.getHostAddress());
+    setPeer(span, remoteAddress.getHostName(), remoteAddress.getHostAddress());
+    return span;
+  }
 
-    String peerService = mapToPeer(hostName);
-    if (peerService == null) {
-      peerService = mapToPeer(remoteAddress.getHostAddress());
+  public static void setPeer(final Span span, String peerName, String peerIp) {
+    assert span != null;
+    if (peerName != null && !peerName.equals(peerIp)) {
+      SemanticAttributes.NET_PEER_NAME.set(span, peerName);
+    }
+    if (peerIp != null) {
+      SemanticAttributes.NET_PEER_IP.set(span, peerIp);
     }
+    String peerService = mapToPeer(peerName);
+    peerService = (peerService == null) ? mapToPeer(peerIp) : peerService;
```

Comment:
i think this reads a little clearer:
```suggestion
    if (peerService == null) {
      peerService = mapToPeer(peerIp);
    }
```